mesheryctl/internal/cli/root/mesh/deploy.go

```diff
@@ -2,140 +2,56 @@ package mesh
-	deployCmd = &cobra.Command{
-		Use:   "deploy",
-		Short: "Deploy a service mesh to the Kubernetes cluster",
-		Args:  checkArgs(1),
-		Long:  `Deploy a service mesh to the connected Kubernetes cluster`,
-		Example: `
-// Deploy a service mesh from an interactive on the default namespace
+var deployCmd = &cobra.Command{
+	Use:   "deploy [mesh-name]",
+	Short: "Deploy a service mesh",
+	Long:  "Deploy a service mesh in the connected Kubernetes cluster",
+	Example: `
+// Deploy a service mesh interactively:
 mesheryctl mesh deploy
 
-// Deploy Linkerd mesh on a specific namespace
-mesheryctl mesh deploy --adapter meshery-linkerd --namespace linkerd-ns
-
-// Deploy Linkerd mesh and wait for it to be deployed
-mesheryctl mesh deploy --adapter meshery-linkerd --watch
+// Deploy Linkerd in a specific namespace:
+mesheryctl mesh deploy linkerd --namespace linkerd-ns
 
-! Refer below image link for usage
-* Usage of mesheryctl mesh deploy
-# ![mesh-deploy-usage](/assets/img/mesheryctl/deploy-mesh.png)
-		`,
-		PreRunE: func(cmd *cobra.Command, args []string) error {
-			log.Infof("Verifying prerequisites...")
-			mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
-			if err != nil {
-				log.Fatalln(err)
-			}
-
-			if len(args) < 1 {
-				meshName, err = validateMesh(mctlCfg, "")
-			} else {
-				meshName, err = validateMesh(mctlCfg, args[0])
-			}
-			if err != nil {
-				log.Fatalln(err)
-			}
-
-			if err = validateAdapter(mctlCfg, meshName); err != nil {
-				// ErrValidatingAdapter
-				log.Fatalln(err)
-			}
-			return nil
-		},
-		RunE: func(cmd *cobra.Command, args []string) error {
-			mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
-			if err != nil {
-				log.Fatalln(err)
-			}
+// Deploy Linkerd and wait for it to be deployed:
+mesheryctl mesh deploy linkerd --watch`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		log.Infof(fmt.Sprintf("Deploying %s...", meshName))
+		_, err = sendOperationRequest(mctlCfg, strings.ToLower(meshName), false)
+		if err != nil {
+			log.Fatalln(err)
+		}
 
-			_, err = sendDeployRequest(mctlCfg, meshName, false)
+		if watch {
+			_, err = waitForDeployResponse(mctlCfg, "mesh is now installed")
 			if err != nil {
 				log.Fatalln(err)
 			}
+		}
 
-			if watch {
-				log.Infof("Verifying Operation")
-				_, err = waitForDeployResponse(mctlCfg, "mesh is now installed")
-				if err != nil {
-					log.Fatalln(err)
-				}
-			}
-
-			return nil
-		},
-	}
-)
-
-func init() {
-	deployCmd.Flags().StringVarP(&adapterURL, "adapter", "a", "meshery-istio:10000", "Adapter to use for installation")
-	deployCmd.Flags().StringVarP(&namespace, "namespace", "n", "default", "Kubernetes namespace to be used for deploying the validation tests and sample workload")
-	deployCmd.Flags().StringVarP(&utils.TokenFlag, "token", "t", "", "Path to token for authenticating to Meshery API")
-	deployCmd.Flags().BoolVarP(&watch, "watch", "w", false, "Watch for events and verify operation (in beta testing)")
```

Comment:
Need to check...

Reply:
What needs to be checked? Please include more details so I can know how to respond or what changes need to be made.

Reply:
I meant the flags... Nothing much